lib/history/memory/inmemory.go

```diff
@@ -102,3 +112,16 @@ func (t *Timeline) getFilteredEvents(params map[string]string) (events []*pb.Tim
 	// TODO: for now just return the unfiltered events.
 	return t.events
 }
+
+// filterDuplicates removes duplicate events.
+func (t *Timeline) filterDuplicates() {
+	set := make(map[*pb.TimelineEvent]struct{})
+	filteredEvents := make([]*pb.TimelineEvent, 0, t.capacity)
+	for _, event := range t.events {
+		if _, ok := set[event]; !ok {
```

Comment:
If the inmemory timeline would hold a map of events instead of a slice, you wouldn't need to filter anything in the first place.

Reply:
Looks like it might be worth to leave it as a slice. We are removing the oldest events if the timeline reaches max capacity. Doing this with a slice is fairly simple. 

If we were storing events in a map, we will need to iterate over the events to find the oldest events.
We can optimize by using a combination of a linked list and a map, but that seems unnecessary right now since this implementation is really only meant to be used for dev/testing.